### PR TITLE
chore: Fix (almost) all the linter warnings

### LINKED ISSUE #53: Clean up code to remove lint warnings

### PR DESCRIPTION
**Pull request by YagoGG**
This resolves the issues in our codebase that make the linter unhappy.

The only ones that I have not fixed are related to the app's theme/colours, mostly because they would involve removing the theme and we are going to need it for the visual revamp.

Fixes #53.

### CODE REVIEW

**Review by sydneyhauke — APPROVED** [review]
Nothing strange here.
Indeed in the future we must be more cautious with hardcoded strings in the app.

**Comment by sydneyhauke** [comment]
These conflicts on the FirebaseUIActivity class... may be a result of the recent merging of PR related to issue #50 ?

**Comment by YagoGG** [comment]
I just updated the PR with the latest `master` and solved the merge conflicts. Thanks for both the review and the heads up, @sydneyhauke!

**Comment by VPellenc** [comment]
I got `AAPT: error: resource drawable/ic_launcher_foreground (aka com.github.steroidteam.todolist:drawable/ic_launcher_foreground) not found.` while trying to test in on the emulator.

Following the advice here: https://stackoverflow.com/questions/49420636/drawable-ic-launcher-foreground-not-found-how-to-find-that-xml-file
I added a new line to the file and that made it work. I don't know if anyone else has the same issue but if so a change is needed I guess

**Review by VPellenc — CHANGES_REQUESTED** [review]
Github is not so user friendly. :)

**Review by lgraziano — APPROVED** [review]
Everything looks fine, I think we can merge it right away 👍

File: app/src/main/res/layout/layout_task_item.xml [hunk 1]
```diff
@@ -8,7 +8,7 @@
 
     <CheckBox
         android:id="@+id/layout_task_checkbox"
-        android:layout_width="wrap_content"
+        android:layout_width="0dp"
         android:layout_height="wrap_content"
```
[the remark this answers is not in the document]
Comment (YagoGG) [reply app/src/main/res/layout/layout_task_item.xml:12]: Correct!

**Review by VPellenc — APPROVED** [review]
The problem is solved.

**Pull request merged**